File: BRU-Avtopark-Avtobusov/TicketSalesApp.Core/Models/VacationRequest.cs
// Core/Models/VacationRequest.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
#if MODERN
using System.Text.Json.Serialization;
#elif WINDOWSXP
using Newtonsoft.Json;
#endif

namespace TicketSalesApp.Core.Models
{
    /// <summary>
    /// Represents employee vacation and leave requests
    /// Essential for workforce planning in transportation industry
    /// </summary>
    public class VacationRequest
    {
        [Key]
        public long RequestId { get; set; }

        [Required]
        [ForeignKey("Employee")]
        public long EmployeeId { get; set; }

        [Required]
        public DateTime StartDate { get; set; }

        [Required]
        public DateTime EndDate { get; set; }

        [Required]
        [MaxLength(50)]
        public string VacationType { get; set; } // Annual, Sick, Unpaid, etc.

        [MaxLength(2000)]
        public string Reason { get; set; }

        [Required]
        [MaxLength(50)]
        public string Status { get; set; } // Pending, Approved, Rejected, Cancelled

        [ForeignKey("ApprovedBy")]
        public long? ApprovedByUserId { get; set; }

        public DateTime? ApprovalDate { get; set; }

        [MaxLength(2000)]
        public string ApprovalNotes { get; set; }

        public int DaysRequested { get; set; }

#if MODERN
        [JsonIgnore]
        public Employee? Employee { get; set; }

        [JsonIgnore]
        public User? ApprovedBy { get; set; }
#else
        [JsonIgnore]
        public Employee Employee { get; set; }

        [JsonIgnore]
        public User ApprovedBy { get; set; }
#endif

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }
    }
}

File: TicketSalesApp_UI_Rust_Linux/src/models/vacation_request.rs
use chrono::{DateTime, NaiveDate, Utc};
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "PascalCase")]
pub struct VacationRequest {
    pub request_id: i64,
    pub employee_id: i64,
    pub start_date: NaiveDate,
    pub end_date: NaiveDate,
    pub vacation_type: String,
    pub reason: Option<String>,
    pub status: String,
    pub approved_by_user_id: Option<i64>,
    pub approval_date: Option<DateTime<Utc>>,
    pub approval_notes: Option<String>,
    pub days_requested: i32,
    pub created_at: DateTime<Utc>,
    pub updated_at: Option<DateTime<Utc>>,
}

impl VacationRequest {
    pub fn is_pending(&self) -> bool {
        self.status == "Pending"
    }
    
    pub fn is_approved(&self) -> bool {
        self.status == "Approved"
    }
    
    pub fn is_rejected(&self) -> bool {
        self.status == "Rejected"
    }
}
